src/Installer.php

```diff
@@ -255,6 +314,35 @@ private function downloadFile(HttpClient $httpClient, $url, $targetPath)
     }
 
     /**
+     * Downloads the available puli versions.
+     *
+     * @param HttpClient $httpClient The client to use.
+     * @param string     $url        The URL to download.
+     *
+     * @return array|null The available versions, null if the download failed.
+     */
+    public function downloadVersions(HttpClient $httpClient, $url)
+    {
+        ErrorHandler::register();
+
+        if (!($versions = $httpClient->download($url))) {
+            $this->error(sprintf(
+                'Download failed: %s',
+                implode(PHP_EOL, ErrorHandler::getErrors())
+            ));
+
+            return null;
+        }
+
+        $versions = json_decode($versions);
+        usort($versions, 'version_compare');
+
+        ErrorHandler::unregister();
```

Comment:
IMO it's a bit confusing and error-prone that this method returns an array *or* null. I'd add this here:

```php
if (ErrorHandler::hasErrors()) {
    throw new RuntimeException(sprintf(
        'Could not download %s: %s',
        $url,
        "\n".implode("\n", ErrorHandler::getErrors())
    ));
}
```